stan/math/prim/scal/prob/normal_rng.hpp

```diff
@@ -13,26 +17,42 @@
 namespace stan {
   namespace math {
 
-    template <class RNG>
-    inline double
-    normal_rng(double mu,
-               double sigma,
+    template <typename T_loc, typename T_scale, class RNG>
+    inline typename promote_vector<T_loc, T_scale>::type
+    normal_rng(const T_loc &mu,
+               const T_scale &sigma,
                RNG& rng) {
       using boost::variate_generator;
       using boost::normal_distribution;
 
       static const char* function("normal_rng");
 
+      scalar_seq_view<T_loc> mu_vec(mu);
+      scalar_seq_view<T_scale> sigma_vec(sigma);
+
       check_finite(function, "Location parameter", mu);
       check_not_nan(function, "Location parameter", mu);
       check_positive(function, "Scale parameter", sigma);
       check_not_nan(function, "Scale parameter", sigma);
+      if (mu_vec.size() > 1 && sigma_vec.size() > 1)
+        check_size_match(function,
+                         "Location parameter", mu_vec.size(),
+                         "Scale Parameter", sigma_vec.size());
 
-      variate_generator<RNG&, normal_distribution<> >
-        norm_rng(rng, normal_distribution<>(mu, sigma));
-      return norm_rng();
-    }
+      size_t N = max_size(mu, sigma);
+
+      typename promote_vector<T_loc, T_scale>::type output(N);
```

Comment:
Yeah, haha, good catch. I should have put a note here. I'd like to not have this in the final code. I agree the better way to do this would be something like VectorBuilder (to not mislead people with sneaky code).